src/bot.ts

```diff
@@ -100,22 +100,52 @@ const onMessage = async (ctx: OnMessageContext) => {
       return voiceMemo.onEvent(ctx).catch((e) => payments.refundPayment(e, ctx, price));
     }
   }
+  if (openAiBot.isSupportedEvent(ctx)) {
+    const price = openAiBot.getEstimatedPrice(ctx);
+    if (price > 0) {
+      await ctx.reply(`Processing withdraw for ${price.toFixed(2)}¢...`);
+    }
+    const isPaid = await payments.pay(ctx, price);
+    if (isPaid) {
+      return openAiBot.onEvent(ctx).catch((e) => payments.refundPayment(e, ctx, price));
+    }
+  }
+  if (conversationHandler.isSupportedEvent(ctx)) {
+    const price = conversationHandler.getEstimatedPrice(ctx);
+    if (price > 0) {
+      await ctx.reply(`Processing withdraw for ${price.toFixed(2)}¢...`);
+    }
+    const isPaid = await payments.pay(ctx, price);
+    if (isPaid) {
+      return conversationHandler.onEvent(ctx).catch((e) => payments.refundPayment(e, ctx, price));
+    }
+  }
+  if (oneCountryBot.isSupportedEvent(ctx)) {
+    const price = oneCountryBot.getEstimatedPrice(ctx);
+    if (price > 0) {
+      await ctx.reply(`Processing withdraw for ${price.toFixed(2)}¢...`);
+    }
+    const isPaid = true // await payments.pay(ctx, price);
```

Comment:
Why `true` is hardcoded here?
If this module is free to use, just remove payments logic or return `0` from `getEstimatedPrice`

Reply:
Fixed (was hardcoded because when I was testing it, I had not implemented getEstimatedPrice)